app/controllers/concerns/commit_matchable.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module CommitMatchable
+  extend ActiveSupport::Concern
+
+  def commit_matches?(regex)
+    params[:commit]&.downcase&.match(regex).present?
```

Comment:
I'm still not sure I like doing this...but I still can't find another way to change what happens on form submission based on which submit button the user presses.

It means we're just relying on the button text, which could be changed for multiple reasons (and we'd assign it 1 story point in planning 😆). Also I know it's unlikely for this project, but it would break if we started supporting multiple languages.

Reply:
Not sure I fully understand the problem. I think you want buttons having different results depending on which button you press??

I assume you've tried things like changing the [name](https://stackoverflow.com/a/42377507/1511504) - [this is similar](https://stackoverflow.com/a/3086833/1511504). 
Saw something where they had [different formactions](https://stackoverflow.com/a/32536941/1511504) 

Any good?

Reply:
Ugh, I'm torn. `params[:commit]` not ideal because of tightly coupling actions and views. `formaction` potentially not ideal because it pushes us towards having eg two different `create` actions for the same model (the 'save' and the 'save and mark as complete' actions). I'm going to merge this as is for now, and then agonise some more about it 🙈 😰